pkg/otelcollector/infra-meter/infra-meter.go

```diff
@@ -231,3 +257,68 @@ func updateK8sAttributesProcessor(processor interface{}, selectorsList []interfa
 		}
 	}
 }
+
+// processMapForSecrets processes the given map to search for the secret reference and replace it with actual value.
+func processMapForSecrets(config map[string]interface{}, secretManagerClient *secretmanager.SecretManagerClient) error {
+	for key, value := range config {
+		switch v := value.(type) {
+		case map[string]interface{}:
+			if err := processMapForSecrets(v, secretManagerClient); err != nil {
+				return fmt.Errorf("error processing map for key %s: %w", key, err)
+			}
+		case string:
+			if strings.HasPrefix(v, "${") && strings.HasSuffix(v, "}") {
```

Comment:
I wonder if this shouldn't be a regexp, instead of assuming that the secret must be a full string. I could imagine that we'd like to allow users to specify HTTP headers like: `X-Authorization: Bearer ${secret_manager.bearer_token}`

Reply:
I followed the OTEL's way of reading the environment variables as I don't think they support combining string with variable. I can check again